### PR TITLE
Expose Coxeter and dual Coxeter numbers in RootSystem class

### PR DESCRIPTION
**Pull request by skipgaribaldi**
Makes the Coxeter number and dual Coxeter number, which are properties of a root system, available in the RootSystem class.

The underlying functions already exist in the CartanType, so this is a matter of exposing them in the RootSytem class.

### :memo: Checklist

- [X ] The title is concise and informative.
- [X ] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ X] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

**Comment by fchapoton** [comment]
you should not touch ruff.toml

**Comment by skipgaribaldi** [comment]
> you should not touch ruff.toml

Of course!  Sorry.  This is my first pull request.  Fixed

File: src/sage/combinat/root_system/root_system.py [hunk 1]
```diff
@@ -787,6 +787,56 @@ def coambient_space(self, base_ring=QQ):
...
+        For an irreducible finite root system, reports the Coxeter number.
+        This function is a wrapper for the corresponding function in CartanType.
+
+        EXAMPLES::
+
+            sage: rt = RootSystem(['C', 5])
+            sage: rt.coxeter_number()
+            10
+
+        .. SEEALSO:: :meth:`~sage.combinat.root_system.cartan_type.CartanType_standard_finite.coxeter_number`
```
Comment (tscrim) [inline src/sage/combinat/root_system/root_system.py]:
```suggestion
        """
        Return the Coxeter number of an irreducible finite root system.

        .. SEEALSO::

            :meth:`~sage.combinat.root_system.cartan_type.CartanType_standard_finite.coxeter_number`.

        EXAMPLES::

            sage: rt = RootSystem(['C', 5])
            sage: rt.coxeter_number()
            10
```

File: src/sage/combinat/root_system/root_system.py [hunk 2]
```diff
@@ -787,6 +787,56 @@ def coambient_space(self, base_ring=QQ):
...
+        """
+        # Check if RootSystem is finite and irreducible
+        if self.is_finite() and self.is_irreducible():
+            # Hand over to CartanType method
+            return self._cartan_type.coxeter_number()
+        else:
+            raise ValueError("Coxeter number is defined only for finite and irreducible root systems.")
```
Comment (tscrim) [inline src/sage/combinat/root_system/root_system.py]:
```suggestion
        if not (self.is_finite() and self.is_irreducible()):
            raise ValueError("the Coxeter number is defined only for finite and irreducible root systems")
        # Hand over to CartanType method
        return self._cartan_type.coxeter_number()
```
There isn't a need to give implementation details in the doc. I find this style of code a bit easier to understand and maintain.

File: src/sage/combinat/root_system/root_system.py [hunk 3]
```diff
@@ -787,6 +787,56 @@ def coambient_space(self, base_ring=QQ):
...
+
+    def dual_coxeter_number(self):
+        """
+        For an irreducible finite root system, reports the dual Coxeter number,
+        which is defined to be 1 plus the sum of the coefficients of simple roots
+        in the highest short root of the dual root system.  This function is a
+        wrapper for the corresponding function in CartanType.
```
Comment (tscrim) [inline src/sage/combinat/root_system/root_system.py]:
```suggestion
    def dual_coxeter_number(self):
        """
        Return the dual Coxeter number of a irreducible finite root system.

        The dual Coxeter number is equal to 1 plus the sum of the coefficients
        of simple roots in the highest short root of the dual root system.
```

File: src/sage/combinat/root_system/root_system.py [hunk 4]
```diff
@@ -787,6 +787,56 @@ def coambient_space(self, base_ring=QQ):
...
+
+        The dual Coxeter number is not the same concept as the Coxeter number
+        of the dual root system::
+
+            sage: rt.dual
+            Dual of root system of type ['C', 5]
+            sage: rt.dual.coxeter_number()
+            10
+
+        .. SEEALSO:: :meth:`~sage.combinat.root_system.cartan_type.CartanType_standard_finite.dual_coxeter_number`
```
Comment (tscrim) [inline src/sage/combinat/root_system/root_system.py]: Move the `.. SEEALSO::` before the `EXAMPLES::`.

File: src/sage/combinat/root_system/root_system.py [hunk 5]
```diff
@@ -787,6 +787,56 @@ def coambient_space(self, base_ring=QQ):
...
+        """
+        # Check if RootSystem is finite and irreducible
+        if self.is_finite() and self.is_irreducible():
+            # Hand over to CartanType method
+            return self._cartan_type.dual_coxeter_number()
+        else:
+            raise ValueError("Dual Coxeter number is defined only for finite and irreducible root systems.")
```
Comment (tscrim) [inline src/sage/combinat/root_system/root_system.py]:
```suggestion
        # Check if RootSystem is finite and irreducible
        if not (self.is_finite() and self.is_irreducible()):
            raise ValueError("the dual Coxeter number is defined only for finite and irreducible root systems")
        # Hand over to CartanType method
        return self._cartan_type.dual_coxeter_number()            
```

**Review by tscrim — CHANGES_REQUESTED** [review]
Welcome to SageMath development. Thank you for the PR. Here are some suggestions. Once they are done, then I will set a positive review.

File: src/sage/combinat/root_system/root_system.py (hunk 4 above)
Comment (skipgaribaldi) [reply src/sage/combinat/root_system/root_system.py]: Done!  Thanks for your feedback

**Comment by tscrim** [comment]
Thank you. Once the bots come back green, I will set a positive review.

In the future, if you could refrain from using GH's "implement suggestion" (or whatever it is exactly called) and make a proper commit, or at least give more precise commit messages, that would be appreciated. The generic "Update XYZ" makes it a bit harder if we have to dig into the history.

Thank you for the improvements! Please feel free to cc me on any PRs that are related to root systems, Lie groups/algebras, representation theory. etc. Feel free to contact me directly by email (see my GH profile) to discuss things such as missing features or issues.

File: src/sage/combinat/root_system/root_system.py [hunk 6]
```diff
@@ -787,6 +787,55 @@ def coambient_space(self, base_ring=QQ):
...
+            Dual of root system of type ['C', 5]
+            sage: rt.dual.coxeter_number()
+            10
+        """
+        # Check if RootSystem is finite and irreducible
+        if not (self.is_finite() and self.is_irreducible()):
+            raise ValueError("the dual Coxeter number is defined only for finite and irreducible root systems")
+        # Hand over to CartanType method
+        return self._cartan_type.dual_coxeter_number()            
+
```
Comment (tscrim) [inline src/sage/combinat/root_system/root_system.py]:
```suggestion
        # Hand over to CartanType method
        return self._cartan_type.dual_coxeter_number()

```
Whoops, we got some trailing whitespace (my fault from my suggestion I believe).

Comment (skipgaribaldi) [reply src/sage/combinat/root_system/root_system.py]: Thanks!  Working on fixing

Comment (skipgaribaldi) [reply src/sage/combinat/root_system/root_system.py]: Got it

**Review by tscrim — APPROVED** [review]
Thank you. LGTM.

**Pull request merged**